### PR TITLE
Issue #2325 - Allow form limits to be set via context-params

### PR DESCRIPTION
**Pull request by mmacret**
I assumed the context xml file would be checked first and if maxFormContentSize or maxFormKeys are not set, web.inf will be checked next.

### CODE REVIEW

File: jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java [hunk 1]
```diff
@@ -97,6 +103,53 @@ public void testSessionListeners ()
...
+
+        // Load a web.xml setting org.eclipse.jetty.server.Request.maxFormContentSize to 100
+        File webMaxForm = MavenTestingUtils.getTestResourceFile("web_maxform.xml");
+        context.getMetaData().setWebXml(Resource.newResource(webMaxForm));
+
+        Server server = new Server();
+        context.setServer(server);
+        server.setHandler(context);
+
+        context.setSessionHandler(new SessionHandler()
```
Comment (joakime) [inline jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java]: Having this as a SessionHandler seems odd.
Perhaps a `context.addServlet()` with a servlet would be a better fit (testing wise)

**Comment by janbartel** [comment]
@mmacret thanks for bringing this to our attention. I actually think this area needs some more work:
1. our documentation says that we support a system property of "org.eclipse.jetty.server.Request.maxFormContentSize" and "org.eclipse.jetty.server.Request.maxFormKeys" but I don't see anywhere in the code where that property is checked.
2. The code to work out how the maxFormContentSize and maxFormKeys is set shouldn't be in Request at all, but rather worked out once when the context starts, so the Request can just call context.getMaFormXXX().

**Comment by gregw** [comment]
@mmacret can you tell us why you cannot use the currently available `WEB-INF/jetty-web.xml` to meet your requirements?

**Comment by mmacret** [comment]
@gregw I am not actually looking to use that feature. I am using Jetty in my daily job and I just wanted to contribute back to the project! I was looking for a easy task to complete to get me started and I found that one. Let me know if you still want some help on this issue or if you have some other issues you'd prefer me to work on!

**Comment by gregw** [comment]
@mmacret Awesome!   So you have already contributed to the project even if this PR does not get merged.  By creating it and making us look at particular code and options around it, you've made us realize that there is some cleanup work to do in this area AND probably that #2325 can be closed.   Standby and I'll make some suggestion in the next day or so of what issues that I think are good low hanging fruit to get started on.... or you could work on the 1. and 2. points that @janbartel  raised above?

**Comment by mmacret** [comment]
Thanks for the feedback! I'll start working on what @janbartel suggested.

**Pull request closed without merging**